### PR TITLE
Improve test

### PR DESCRIPTION
**Pull request by Milly**
Improve test typing.
Add test function option comments.
Add shared-server test helper.
Show verbose logs use `-V1` instead `redir`.
Error stack trace improved to easily understood.
Add `plugin/denops.vim` tests.

## Summary by CodeRabbit

- **New Features**
  - Added support for specifying run mode (`vim` or `nvim`) in test utilities.
  - Enhanced wait function with improved timeout handling and custom error messages.
  - Introduced functionality for starting a shared server for testing purposes.

- **Improvements**
  - Test commands now include language settings for consistent output.
  - Expanded options for verbosity, prelude/postlude commands, and environment variables in test utilities.

- **Bug Fixes**
  - Improved error handling in test utilities to provide more informative messages.

### CODE REVIEW

**Comment by Milly** [comment]
@coderabbitai review

File: denops/@denops-private/testutil/with.ts [hunk 1]
```diff
@@ -6,24 +6,32 @@ const script = new URL("./cli.ts", import.meta.url);
...
+  /** Print Vim messages (echomsg). */
+  verbose?: boolean;
+  /** Vim commands to be executed before the startup. */
   prelude?: string[];
+  /** Vim commands to be executed after the startup.  */
   postlude?: string[];
+  /** Environment variables.  */
   env?: Record<string, string>;
-};
+}
```
[the remark this answers is not in the document]
Comment (Milly) [reply denops/@denops-private/testutil/with.ts:21]: The string was too complex so I split it into multiple lines.

**Comment by Milly** [comment]
@coderabbitai review

File: denops/@denops-private/testutil/host.ts [hunk 2]
```diff
@@ -3,16 +3,17 @@ import { Neovim } from "../host/nvim.ts";
 import { Vim } from "../host/vim.ts";
 import { withNeovim, WithOptions, withVim } from "./with.ts";
```
[the remark this answers is not in the document]
Comment (Milly) [reply denops/@denops-private/testutil/host.ts:4]: These are class instead types.

**Review by lambdalisue — APPROVED** [review]
LGTM

**Pull request merged**